diffkemp/simpll/ControlFlowGraphUtils.h

```diff
@@ -0,0 +1,106 @@
+//===------- ControlFlowGraphUtils.h - Utility functions for slicing -------==//
+//
+//       SimpLL - Program simplifier for analysis of semantic difference      //
+//
+// This file is published under Apache 2.0 license. See LICENSE for details.
+// Author: Tatiana Malecova, t.malecova@gmail.com
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains declaration of utility functions and sets for slicing a
+/// function.
+///
+//===----------------------------------------------------------------------===//
+#ifndef DIFFKEMP_CONTROLFLOWGRAPHUTILS_H
+#define DIFFKEMP_CONTROLFLOWGRAPHUTILS_H
+
+#include "Config.h"
+#include "DebugInfo.h"
+#include "DifferentialFunctionComparator.h"
+#include "Result.h"
+#include "ResultsCache.h"
+#include "SourceCodeUtils.h"
+#include "Utils.h"
+#include <llvm/IR/Module.h>
+#include <queue>
+#include <set>
+
+using namespace llvm;
+
+class ControlFlowSlicer {
+  public:
+    GlobalVariable *Variable = nullptr;
+    // Instructions directly dependent on the parameter
```

Comment:
If I understand it correctly this doesn't check dependency on the parameter, but on the differing instructions: the comments should reflect that (not only here, but in the whole file).